javalib/src/main/scala/java/util/Base64.scala

```diff
@@ -0,0 +1,486 @@
+        dst.put(table((bits >>> 18) & 0x3f))
+        dst.put(table((bits >>> 12) & 0x3f))
+        if(dst.hasRemaining) dst.put(table((bits >>> 6) & 0x3f))
+        if(dst.hasRemaining) dst.put(table(bits & 0x3f))
+        currentLine += 4
+        if(lineSeparator.length > 0 && lineLength > 0 && currentLine == lineLength && dst.hasRemaining) {
+          lineSeparator.foreach(dst.put)
+          currentLine = 0
+        }
+      }
+
+      while(src.remaining >= 3) encode(src.get, src.get, src.get)
+
+      length % 3 match {
+        case 0 =>
+        case 2 =>
+          encode(src.get, src.get, 0)
+          if(withPadding) {
+            dst.position = dst.position-1
+            dst.put('='.toByte)
+          }
+        case 1 =>
+          encode(src.get, 0, 0)
+          if(withPadding) {
+            dst.position = dst.position-2
+            dst.put('='.toByte)
+            dst.put('='.toByte)
+          }
+      }
+      dst.position
+    }
+
+    private def dstLength(srcLength: Int): Int = {
+      val withPad = ((srcLength + 2) / 3) * 4
+      val toRemove = if (withPadding) 0 else (3 - (srcLength % 3)) % 3
+      val withoutEndLines = withPad - toRemove
+      withoutEndLines + (if (lineLength > 0) ((withoutEndLines-1) / lineLength) * lineSeparator.length else 0)
+    }
+  }
+
+  // --------------------------------------------------------------------------
+
+  private class EncodingOutputStream(out: OutputStream,
+                                     table: Array[Byte],
+                                     lineLength: Int,
+                                     lineSeparator: Array[Byte],
+                                     withPadding: Boolean) extends FilterOutputStream(out) {
+
+    override def write(b: Int): Unit = write(Array(b.toByte), 0, 1)
+
+    private val inputBuf = new Wrapper[Byte](new Array[Byte](3))
+    private var currentLineLength = 0
+    private var closed = false
+
+    @inline
+    private def addLineSeparators(): Unit = {
+      if (lineSeparator.length > 0 && lineLength > 0 && currentLineLength == lineLength) {
+        out.write(lineSeparator)
+        currentLineLength = 0
+      }
+    }
+
+    @inline
+    private def writeBuffer(count: Int): Unit = {
+      inputBuf.clear()
+      val bits = (inputBuf.get & 0xff) << 16 | (inputBuf.get & 0xff) << 8 | (inputBuf.get & 0xff)
+      var shift = 18
+      for(_ <- 0 until count) {
+        out.write(table((bits >>> shift) & 0x3f))
+        shift -= 6
+        currentLineLength += 1
+      }
+      inputBuf.clear()
+    }
+
+    override def write(bytes: Array[Byte], off: Int, len: Int): Unit = {
+      // scalastyle:off return
+      if (closed)
+        throw new IOException("Stream is closed")
+      else if ((off < 0) || (len < 0) || (off > bytes.length) || ((off + len) < 0) || ((off + len) > bytes.length))
+        throw new IndexOutOfBoundsException
+      else if (len == 0)
+        return
```

Comment:
You can easily get rid of that `return` by doing `if (len != 0)` and putting the rest of the method in that if.